lib/ClangImporter/SwiftDeclSynthesizer.h

```diff
@@ -334,6 +334,14 @@ class SwiftDeclSynthesizer {
                                 const swift::Type &swiftParamTy,
                                 SourceLoc paramLoc);
 
+  /// Synthesize a static factory method for a C++ foreign reference type,
+  /// returning a `CXXMethodDecl*` or `nullptr` if the required constructor or
+  /// allocation function is not found.
+  static clang::CXXMethodDecl *
+  synthesizeStaticFactoryForCXXForeignRef(clang::CXXRecordDecl *cxxRecordDecl,
+                                          clang::ASTContext &clangCtx,
```

Comment:
Let's remove this parameter, since we can always retrieve the context from the first parameter.